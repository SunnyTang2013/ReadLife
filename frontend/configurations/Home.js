import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => {
  return (
    <div>
      <h2 className="display-4">Configurations</h2>
      <div className="row my-4">
        <div className="col-12 col-sm-4 col-md-3 col-xl-2 text-center">
          <Link className="app-link" to="/job-config-group/list">
            <i className="app-link-icon green fa fa-fw fa-cogs" />
            <span className="app-link-title">Job Config Groups</span>
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Home;
