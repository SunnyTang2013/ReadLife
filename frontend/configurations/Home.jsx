AES_ENCRYPTED_FILE
ltjgCdz9te4jKRKIEvyhVWdBQUFBQUJvdkM2NUhWbjB5ZDVLTDk1WVJQNGVXSGkyU2wtajdzd0p5UzExWXFGU1VpNnE5T3hpSTNiUjRpNW5DNlZEVF9ESXp3aGcxdl9peWRLcHlHdThTSHA2UlZuWEJlYTFneU9CaU1TRHRsYkVsa3kycTM1NUVPUFhxajlIcmlpdzZXMHpzdUZ1VXJhZl83LTcwZUMyX2NfYTJvMU5HV3Q0VkxJd2d1SDVyc0tnNDJsa2lhcUhlcTFWQkg5Vkd5VUJCS09Yby1yRi0tVzRocUR2T0l0OXh5TWIzQlJETEVNM3hGbEJqYlliTXZ6VzRJUElFVjI5LTJVQlBhNjNZcVNfN1llN3Z6bXQ4MzVpM3EzbDU1WWpJeWRDU0tXQzZjalhQYVBNM0ZOdlh2NDgwT2ZXSW93Q3ZJMFNnUEZDWE03ckpTN3JIY0NQd0VYYzI4VTdycHpLNG0zdTRQU1c2bFZ2LWZ6cDE2UUZCZ08tQS1pQms5TFBXU2pSSmRGT25lOHZFV1E1YVBrelVPQ3BMSHRuclJxUUpOR0Z0TjdfQlFobGhRYjZ5VDNYbGdkaHZ0QmkxcVVReGR0R1UtMktON0M3OFVjMktvclcydWE1MkdmVWVWc0luTEc3czVuVXlFSXNnTWVVVE5mc3dRZHQ3MEJ3cnVSX3RWQmI2V1dGcWtvSlhVdi05aUlJQkczaDgzRW4zNDdoMFoxYUJsbHhETFhMTmZQMkx6QUhLaFVaM3JwODhiM2xvcEl3c2k2N0c4Zm5XbmdxaFQyQmhSOVl4WWhVT2NENmtGQWNvelVReEdMYlZyZWJEcngzanVaUmZHZUVISXU0TVhQcmJHeDFzWXlHbU5EUVBEUmFiWGpTS3ZrZ1B6Q3plM25Nc2N3Nk5UYWtjU3dvelpBTFY4TnNfcjBZZ09oSUtLdTk4M0ZUX3N2WG9BTmRfNlUzdTJQZElQWXVvOEdBZGxXZWNDSFU2emh2T0VQOVBuMTVKdFJRTkg4dmRfck1mY3ZRMnF1aE1JWGNjNEg2MnkxR0U4T3BOWDgtYVZhOVdLZmplbWlsMUczYUNyU01pVkVXVDZhMnEwZXlSamM9