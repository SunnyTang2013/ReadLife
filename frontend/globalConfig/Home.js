import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => {
  return React.createElement('div', null,
    React.createElement('h2', { className: 'display-4' }, 'Global Configurations'),
    React.createElement('div', { className: 'row my-4' },
      React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
        React.createElement(Link, { className: 'app-link', to: '/job-context/list' },
          <i className="app-link-icon blue fa fa-fw fa-rocket" />,
          React.createElement('span', { className: 'app-link-title' }, 'Job Contexts')
        )
      ),
      React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
        React.createElement(Link, { className: 'app-link', to: '/execution-system/list' },
          <i className="app-link-icon red fa fa-fw fa-server" />,
          React.createElement('span', { className: 'app-link-title' }, 'Execution Systems')
        )
      ),
      React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
        React.createElement(Link, { className: 'app-link', to: '/bank-holiday/list' },
          <i className="app-link-icon yellow fa fa-fw fa-yelp" />,
          React.createElement('span', { className: 'app-link-title' }, 'Bank Holiday')
        )
      )
    )
  );
};

export default Home;