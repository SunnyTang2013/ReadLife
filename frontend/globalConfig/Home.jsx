AES_ENCRYPTED_FILE
ltjgCdz9te4jKRKIEvyhVWdBQUFBQUJvdkM2NWRCMUFSam54VERkVlZ5RC0xdzBrb1l4V3dJQTltMkxBQmFrblctZEpjVXZONnBpSXdBUjcyZm90NmVjcjF2cmxmbUwycTBIS3owcm9RS0RSZVR6QWl2ei1Tb1Q1MjRnd3o2YXlSQTNKM1BhWEZyLUdYdWtvRFdEVkd1UjBFNlFXNFFuZ3Y3SXluWkNhWjdCdFdGTnl1TV9iNFFYdGZKSWhIZkE0bUVhZVJXcDdDdWZmLTlXRlNPUkJJLV9jR21HYnAzbTRkdG1XRmNLZE9yWmZ4SXdyaktNVmVCdXVuLWVEdWtJb3RXM0psdzF6RU02Tk5SaTNycXdrU0VicG1fN2tHdU90ZVo2dGM1WFhEa1VoR2JQVFo0M2R6NVlVbkNIeDZXSHNpd3N2X1RVREVoQUNKU3BOUWlXZXN1MmQxYXhLSmcwZmZrckFhT3JQS2tPalZTR04zaWVHcmc0OHA3Wk9FcVo4X1hEbE0yOEpsc0M1ZkRaTUZBek9PZEpCNzFrQVNJeGRPVkRLRENDMnZNWUlRZ2xDczZwSnljN3BoTjQ5ZE9DMTRwcUsxUXRiQi1oMFRLNG9DdEtsclZRYS11N1pZamVVREoyXzU0dHVtTXEzNWJ4el9EdVliLUtteW83c0RDcnFOLTF2ZXFBVkJibUtKOHRXWWhkNERQMHczSjlGX19fTWdVZ21Wai1mT2l0OE4zUmh5aFU3U1c4dDNSZnllZ2VVYTN5TXhSRy1PTU0xRFc5S2FsVkVjTVF5VVl3V3FUOHJfY2NJS0I4U0xKaGlyZlhFM2kybUxSNEtwN3dNcVdseWN0dzdUdzMxZ3B0bUVOa0wxUllSWlZlU2FWQjh5RFRXT0locFhOMjRsNUVLOFp2UUh3U296angtUGhQY29DTnRDbm9SM2tsV1lSeXY5cmZVS2hUUmZLdV80ekNLWDNqdmJyc01EMjBMbG1XdDlUVUM5VnRTUlg4bVgtYkIwNkdlR2pGSllIQlMxY0poeVJKWTI2aWVxRC1pZGVuZld1c0FxMHQwTDk0bENrdWVha2dPam1iQXlyWkdLeGVPVHBXMHl3NEg2bTY0RkpSMkdZeUNwcVBadUpfdS1nd2ZfNzBLbTl0S1dFQkdOSkZlZ2Y1eGhMWm1GbHFIS1RJZUJPeERkbTBRdFRMd1ZpZE9LbUp0NnQ4MnNqVk11TDJWWklsdEtFcVlTaGdkTmdQczVEb0Q0UnJLOTRQTnRDZ0VibzdIb084Z2pNX1FFQlkyVV8tMnUzdGgwdmxzMTJEQ3ltM2loYXZRVFNmYk9kYTc1NlhNTm5IYzNuWXcyb3pRMjhwbm8xRXpHZVFZUThQNzV5clJCUm9NQURvYlZNQWJEaHY5WVpaLXZZUWV4SmY4VXBERGJjVFZ0em5rYTdKZm9PTWNWb0JjVHlnYzZkUDZyc09wLVV2TWNMRk9uTFJ2eGotMV9oVTFRZ3Fwc2x4cDVxOUVWQ0dqMlFBanczZFlVdjJXT3gzQmVJdXBucE1EX3BobURidzgxcW93ay1hZFA3WDFqeGNxTmVIYjdEQ1JvTXpMQjBmNElvWmhIUmlBRDJTdTdFcmRFMGV2NXdzT21faE1zdmVmSzc5ajItYXZDa1ptdzdCM3pFTUN6WFVIeC1NQm5pbGpoQkViRE04b3duQXZhVm9IZy1oaVZ0RDI5MjlBZmhSNXFWMkVrdFdXdEJidU9xbGd4YnFjUnNEakZXQjVkc1VzOEE2NHR4M0IwNDJuOHZ3Mk1wdTEzeE9WWV80S0xpUUNvWk4zanM5XzBZcXZ4TVlnakYzbUlDeGRWSVpsQnpLZ21hanR6QTloQXJRQ0JhMk9hbGJLQ181UjlUTHBvVTlhTTUxVEc5V3h2LXkzRFh4SElSQXQyMUdxQzNlNGQ0dnpUTHpHR2xHWlBkZUctOEc3eklnVFZKRU94eVd2MVdacGdJOHB2dFg2dWxTMXl6SWR6VEdiaWVocEljWDk0MS1HNFRQR3VORjhaZWwta29zN3BOQi05OTdseFY2ektUbUxFZmdfSGZ4UVRXa3E2TmRfcmlvWFl4TE9JX3Vpb0hRTGN2LXppWldUTXJ6d2k1Y1pRV0VlNHE5ZDNWSFdTbzlLOWxnWm9JaWEtOG1PZGZQM3diQVAwanpVZ19CcHFRPT0=