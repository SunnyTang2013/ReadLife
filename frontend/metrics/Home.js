import React from 'react';
import { Link } from 'react-router-dom';

export default function Home() {
  return React.createElement('div', null,
    <div className="spacer-top" />,
    React.createElement('div', { className: 'shadow p-3 mb-5 bg-white rounded' },
      React.createElement('h2', { className: 'display-4' }, 'Metrics'),
      React.createElement('div', { className: 'row my-4' },
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/Metric' },
            <i className="app-link-icon cyan fa fa-fw fa-bank" />,
            React.createElement('span', { className: 'app-link-title' }, 'Phoenix Metrics')
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/MetricTradeError' },
            <i className="app-link-icon cyan fa fa-fw fa-check-square" />,
            React.createElement('span', { className: 'app-link-title' }, 'Trade Errors Metrics')
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/MetricBubble' },
            <i className="app-link-icon cyan fa fa-fw fa-bullseye" />,
            React.createElement('span', { className: 'app-link-title' }, 'Bubbles Metrics')
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/MetricQtf' },
            <i className="app-link-icon cyan fa fa-fw fa-bar-chart-o" />,
            React.createElement('span', { className: 'app-link-title' }, 'Pipelines Metrics')
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/MetricRerun' },
            <i className="app-link-icon cyan fa fa-fw fa-wrench" />,
            React.createElement('span', { className: 'app-link-title' }, 'Reruns')
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement(Link, { className: 'app-link', to: '/QtfHandover' },
            <i className="app-link-icon cyan fa fa-fw fa-envelope" />,
            React.createElement('span', { className: 'app-link-title' }, 'QtfHandover')
          )
        )
      )
    ),
    React.createElement('div', { className: 'shadow p-3 mb-5 bg-white rounded' },
      React.createElement('h2', { className: 'display-4' }, 'External Metrics'),
      React.createElement('div', { className: 'flex-container' },
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement('a', {
            target: '_blank',
            rel: 'noopener noreferrer',
            href: 'https://lookerstudio.google.com/reporting/6f063e03-c862-4119-868a-2b546d3600ef'
          },
            <i className="fa fa-table externalLink" />,
            <span>Preempted Report (UAT)</span>
          )
        ),
        React.createElement('div', { className: 'col-12 col-sm-4 col-md-3 col-xl-2 text-center' },
          React.createElement('a', {
            target: '_blank',
            rel: 'noopener noreferrer',
            href: 'https://lookerstudio.google.com/reporting/ea53193d-e100-4205-b6d5-22aa6d76c9d9'
          },
            React.createElement('i', { className: 'fa cyan fa-table externalLink', style: { color: 'green' } }),
            <span>Rerun Report (UAT)</span>
          )
        )
      )
    )
  );
}