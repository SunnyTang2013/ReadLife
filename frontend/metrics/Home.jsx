AES_ENCRYPTED_FILE
ltjgCdz9te4jKRKIEvyhVWdBQUFBQUJvdkM2NUduNTVtUWhBTmZDeXRhRWJ3bXlNbnFTMXE3c0p4OURFenJJTkRDMkxRU1ZzZHNMV1dadi14c0VUd0xhWHNPR0hzZUFmU21HLTFnQnd0eExZTXlBQmJaYVpYWU5LLWc3R2F5M0ZYN2pwYUpYVGtqUzFBdTJNRGZLajJDT2J0U0N4TEM1WmFMemRyQ1lMVVpKdVJ0V0t3ajhpanNqMWo1c0hfOUFXQldFQnBwbDE4QXlTbUF3WmxNVzhRRU1KTmVEYVRPMHZ3LXJNekdLdml5VTNPbkhVYUt3ajhSMVJDdkhLUlkyenMtVV82WEVESVBmblNWUGdZcmdpYU9kQ0dZbzdpUzRpOTR0LUxmY081NVFtZW5KWVRuWGVUNHZrZTNITmdCV3ctRUx3QnVMVlVjWE9pRVd5Vl9Ob2VlSVVaTmQ3UHMxRHZHZC11N29oTzZ3WXdkTEFzNjVibFFFdjZlRFNMNlhCZ1dCdS1SZDJGU2pySjBQcFZ4N1UwQW4tOUpVTzJUZ0dBRVdOVm51TFJTbFViVUctd29PYlpLNmYyWUxkNkhFUU4zT3dfUnBoVTFFeHlWTGs2OXJ5QVJjNVZMWnBaS1llMmhaQWtrbVhNV3NsblhwZ1VZM3ZyRTEtbkVVWU5HUlBCekxIZ1V3Z0RhSFhuZjNRSTBKV0Ffa3ZHdy1WdjRlZ0VuclJ0X05yUVI3SkIwbXE1cnZsVHdxQU5FMUpmZzlwSGU5dVBjSVI5SWFKWDZiLU1JQWYzNXhwTzNuQXcyVG5RdEpDb3RkcFhOSWg2RmZUR0wweHpZNjF6QjVMV3UyNWxEMmxlWFpMckJ2cTBoX25haEhpMFpMaVVUZkVCbGtCMTdjTnREZDVTbmdxWkhMZVJsLUlJNU9IdG1ibkpkWEtHWmFjczE0VjdjS0RWMGlHZzdaNlVTb0R3dGQ0S2x2UWI3NEdzR2NMWXJYckF0aE9Rc1NCYjJvRl9xUG9xWHVibzVBUDJKamNTNEpaODNsWk5ULV82TWdlVkcxWjlJQ2ZZclE4SFZJak16SXZveTZIVDIwQnM1Ynh1NkVHMVdZeTNvODBydTNxQVdDMEpFRU5FODUxLV9ubGQ3T2I1RG5UdmpWT2pQTDZvckJBZVN4eGE0WXV4bEI5V25CZVItaENDWnZmLXdTUG1wLVpGbjVpN05abC1CeUxHOXZlM2xrb3huRENBejIyTk1sc081QkdqNlVEUnNRUnlMQzVMcHRxMmRXbUVPUHh0VzRubVBxbHpMZjRocUctMFdzNVB2LTFBYlZEZDd5clhhR21HRmNRdXVKVXFad2k3OHV5MWQ0a1pJQ2VZcnljQ19qYnFlYms0RTNjVFZubmZweTU3czhNeWJTVE5vNDYyTmtmb3puLXpGbU9pc0NOckZlVjBDdm5sdVBmS21jd1YteGFtTU9rTlJuRm5yZjItNEFIT3hEeUhnZ2NzZjBvM3RPblhYZUsxTjVBUl9oY0tibG8tRzNGa2hMQ3VnQTIxUWw4WUNRTWYxdlVoX2FLTzk2bkFvZ1VUTDdRdDlaNXJTQWR5Q2t2QkVnRm1TODdCSnc5YTlPYmNBcGR1UWo1cTdOVG5PT3RSdmczeHRiMFdKOUJjcHhDSU1Pc1pWZmpTdDhCYl9XTVJVRjV0cGVrLWFkQ3dCYTNRaG1NRU96bmtrZngwcWVhdnRDT1o2WEdycmhId3RDaHlfN3llQURJenBpNlJCS2QzR0hXOS01UnBmMGJ6WmUzbTZoendmeDBfeFlpOFA5YTlZemFKdnJ1RFF3c0RrMUt2U2hGS2VhTDlPdWlPWXNyRTU3aElJMGtFeDRaZG9xV3EtSEkyMHlCOF9RZU9vSDA3SUJxWmVycVhPTVpBb2ZHODF6NVpGb3h5LWMtNTFaRDFSYk1ZM2g4eWZIajA1cDJGS1pQTTJOVlRiSFdjcm1mWl9sY0xwQ0xCSmQ1TXpfd19NbUdqYXZGUXlGU0pNUkxKOXJYTFZXbXVWMmpVLXh2dHBKdEpYOXAzZzU0My1zYjIwV000eFdwdFo2ZUxaVnZaMnpoMXQwbWdOZGJZTzB1bHduaGJoUXktWkZFQjYxbXpyNkk0N2pVdkUwS2hDTEVFc1hvaDgwOVVEb3dZS2hocU9pNnM1cklxMjhCcmtvb3RFMGJmVWd3VXJaMUtOY0NaMTJ0NGp5ZEs3MHZ0QXc1cjljNlR6N0VxYWI3c0ZyUXROcGhtSENOSEpOcnNFVGh5MUhCLUdQWEk5UFFsQ2RnRkN0LUFRX0JnaF9lNnFoWXE1TGluTG1NZ3dFTW1oWW0zTjViajB6SHpKMzFlbEhGMmNwdDc2alFpbEhkczkxREZqSlZFNnNSX21iZXl3NFlWUTl1OE9hazVQblNCby1FNy1IN3J6UExEWHNfeVExSlg0MmFTWGthMmJkc2QyZE1xcU1ybks3NnJTdk9wQncwUFNTczJudnNvYjNkZzVrQjFKelVnSHBqeFpnTUMtSzVHRklTRks3R09qdFRKam5fOWpHY2ttTVNJbTRhWUV1RHhpTjAxYXJybVk5T05xTERJVXJ5dWttNlphZDAxNE52cWhONlR6bkszQzJDa3AxVHNUeHBUa3BFQW5Na21venEzVUdRWVBZUTh0d0R3bl9CVnkzeDhWT2pxcWhka0Z4Zko0S2dLRFZkNjVPZkJJZEpaT0tUU0d4MV9VRjNuV20zSFB6aHd2OF9HLUFZQ1lMQVdfalh1ZXNKVnFTQm8yeFVkQ2lIWGRlWFYzVzV5UmdHbGwyR01OLXY3MGFPUzZYNlhVeUtCa3ctZ3VPWEZzSnY0V1pRbnZzblBzODZCdElLSXJmUU12Zk1sR2dGcm4xNnV5N21tUXZ0V2NWeE53RnlvMENOeGlKWHZ1VGZ0M1FNbW9UUlFIa2ZLcWRtSDFEUXh1RVZybjFEcjF0UG1iOHJPSFJ1NTdKLTY5TWpfZDBidmt1VkJmNDYwa1JvMy0ycm42SVRqcDRud3NWWDhQN0laT1RXR1BxVERGNjZISVlYV2t2YWxaeDNHRkc0cmlxNUpmOUt3NFVwMEVXVVM5RUlURWk1b2NJcUR0b2lCTjBaQS1WYlhZTU5Zc2k4cWsxRHhNQlNMa1Y3WnpZaEZHdjlDLVZPOWtYM0VUMHMxTFBxRHp4ZEp1b29QN2xPaFZIS2RTRlNwSTJNS1JzcWRoZHlvMmg1SkgyOU5lYUh4bjFHX01xcjZFalZONHVkSXhGY08ySDVXMllkYlhuSGpmSEpUR2NrUXFMNUw4RUV0Ni1qd2Nkb0JDNjB6eUh1RVJtbFEyclVEb0x6QV9NNmNkdE1vYUluRmJGTzdQSmpEVzhhUktoYXB6d1U2dUpxRFRiQVNNaUhMRXBzZDRpRkdUR3lFTXVJd2FBeGVoOFhpTFZPM3o4UTU1a0xmRHE2cXFERGM4UzFicTJhZjl1T0piQ0tVNDgzeXlIR1AxUHlRcFVUODVjR3RCY1o3Ym1WWWFWRE9odHFjOWxlNTVzRzRCWlVOaWVNTy1rU2hsRWtKbmtOMzBzbVdWTDNjN3dBUXNLRDRKdUJicUh6UUpwQnBsWmJkcVBQOTRaNkRDcG50UW5LMWxWV0ZHZHVLNGZFNWZmTWR1N09WSHluMVUtZ29pQjg5SG9rbW1yYXZGNF9hc3hhRDI1QzBwNk9zdDJ6eEx0RTNGOUVxZHVvLVFQZlhHaHRZb1hPZXV0OFhpWGpxcGZVa2E1aTFpcXpsNGJfWURaaDJZcTdZLTNGdVNwc1dPc3ZNX01tSHo0VDc5eDZwMGhkUVJkYlAyQm9WWnpVY2cyYktRelo4U2hNS3lDRUlWTnF4N1pGUzJBNWhPR3FvT3JpRWdWYlBkUzZ3QUZsbV9YdnNnM0JiTnpmRnFnSXR0cXVwcDBIRENBUnJ4Qll5M3otOElvd19OdXN0RG1Na01DQzhCNFdkWkVqUzZUWERaWU1ycnlIRGNhYzlDTWVSTzByUmNjRVVtdmtJVG8wd05zRlRBejY3c1Z5Qko4OHdpb1duNW9pdHduN3JwWlBIQ2dyNVY3anBQbFNTUTJWX01FV1kxNkVDNmZnNlBnMEJGVkNKQTBQMUhWb1RVQTcxYTFWRFA1Q3o4QjNkSzlZNFRDN2lBcUs4dWpGUjY0NldfQUYzRlRPWWZzVk5BaERlcHh6Rk5pSWlSMlQ5MDkyeGc2NzFYU21iWk80TmlEN0pIU1QtQ3RCSnVnSnVGNktsS1E1WjFFM1h0Q1h4b1BJU3QyUTRBTFpQWF9hWDJYelVySkxnMGJFRUFHREY0VkhDcHFoZms1ZlB2N192WXg0cXE2Z2Jsa0ZFSnkycVNzSW13Y1N0ZEtKZHdERzNDUTJ0d0I1UEx6c0NZQ1NSZGI0dkpxcm15aDZwUC0tcDVxbVQxWXdiRTZEWnlNR3U0U2RzbTRNUHZPWXdCbTZ4UnFpZlhDNUJaUVVCVmNGOWJqNVpPN3NYSmNWWFlxWDZGaE5wblFBb1FqdFV2Y2RpVFV2aVhZTVZoa000Z0hLY1dBZl9iMGJVM3dhbUZnc3NGOGd2em5uc3Z0SlJlZmJXeU9lelQzSThZVlFxYTBTY3pmeS0tVDg0bEo2QWRCOWdtRlJYZ3hyaFRJUmRaN3hJTno1eUtVU2tpVWh3dHh0Z05rYWNOclRScmxmRzJBZ0dxSTh6YXB0amxIZU9sbEZ1YTljODk0TWRyRGI4UmlOck9IWmZWYy1tN1dCMlNiWGhkR0pNMk1yRFo0U2pBcVRVUnN4bGotNXpNbTZqdnU3M0VONkFxM05nR3pSdUFmVDd2N193Ul9IZW5yNVJpdFlyTGVSVDNrUFUxU3FWem5JZXhuQ3NhWjYzLVNxUDBOQXBTc3oweXU0ektPTUdBN0xoSFozNll0REU0NzhSRGVtdlpoRWNnWXBVQXI1dlNsdzAyanc0MXlVUzVTb2xPMWRUSU93VlFSUE5KWEVxWjdObVdTdmhQdnc0R2RRQjFmYWwydWRDZkxjWTVlWVBaQWx5ZXUwWlg4amFybDNzMGswMzVuOHhyM19CMmxMb1MyRDEwSGVueEpxUmR3UmxFYkc2ZlZDTlp4MU14VThOWnpFSHdVdm9EUGw2Ti1hWUJyR2ZhLWxDRG03amdYZWRQT3pSdkFtSE5IcGM2Umo0dENsNTJMWHRvcmpac01NeWc2ZVVObFhXbWwzblZ3LVZETExmdHVRWU5FZVRkNzhvc0V1VncxVG9qT3ktMlNBWXdNakhuYW9nSFV6RURhRlg5cjBZMzRPUFUwVlV3aWluMHphNjBDLUlmbUFSSHBydVlXbjZfa0EzYTN5amx1ZDZxZy02X056Sl9wckhvRWFUWDVCNkgzUnpWaXM5bXdVUWxSRUNtUzRYanJyQnJnZXUzSVRKTVFVb2lzLWZOMHR4U1psckNNNlg5aldQYWNZbWRzRDFYNTFkSXVQWG5vQnV1ZmJBTF94SjFfcHFzOHduVGVMb2ZOS0hWRDRONlRVZXI3Q1JpdVY4dWRvWlEyTUt5b0tIR3B0R3dZMUdzcHNlSFNNSFRMTzFTOGc5ZGdOand5V0xvRy1sZE1Ob0doM1Uyc1h0dTRhSFBneWs5ZFFFeTJVR3RMTnNZRWtma1RGTVRPemxCeEJvc3pqajR1X0lYbjBYLTBHZ29KQ2FqaGpUR3JmS2JtcVliNFNyVjByczl4Uk9UM0FBVkQ4alhGb014WTFMN19LaEVZZkVXX0E4Y0tVUDFYUVhqM1BWTmIySUxndUw3VmJkYVFpZmljSy1IWkNQYm1UV2Z6eFRlcEw0M0pZSzFWTHhQS1ZHX095aGVDckdVUUVWdVQxd3VyZkNSRT0=