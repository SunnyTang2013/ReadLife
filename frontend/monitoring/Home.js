import React from 'react';
import { Link } from 'react-router-dom';

export default function Home() {
  return (
    <div>
      <h2 className="display-4">Monitoring</h2>
      <section>
        <div className="alert alert-primary">
          There should be a global dashboard...
        </div>
      </section>
      <div className="row my-4">
        <div className="col-12 col-sm-4 col-md-3 col-xl-2 text-center">
          <Link className="app-link" to="/pipeline-request/list">
            <i className="app-link-icon teal fa fa-fw fa-plug" />
            <span className="app-link-title">Pipeline Requests</span>
          </Link>
        </div>
        <div className="col-12 col-sm-4 col-md-3 col-xl-2 text-center">
          <Link className="app-link" to="/batch-request/list">
            <i className="app-link-icon green fa fa-fw fa-tasks" />
            <span className="app-link-title">Batch Requests</span>
          </Link>
        </div>
        <div className="col-12 col-sm-4 col-md-3 col-xl-2 text-center">
          <Link className="app-link" to="/job-request/list">
            <i className="app-link-icon blue fa fa-fw fa-rocket" />
            <span className="app-link-title">Job Requests</span>
          </Link>
        </div>
      </div>
      <div className="row my-4">
        <div className="col-12 col-sm-4 col-md-3 col-xl-2 text-center">
          <Link className="app-link" to="/execution-system-view">
            <i className="app-link-icon blue fa fa-fw fa-server" />
            <span className="app-link-title">Execution System View</span>
          </Link>
        </div>
      </div>
    </div>
  );
}
