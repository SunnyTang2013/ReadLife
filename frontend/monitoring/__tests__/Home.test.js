import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';

// Helper function to render component with router
const renderWithRouter = (component) => {
  return render(
    <BrowserRouter>
      {component}
    </BrowserRouter>
  );
};

describe('Home Component', () => {
  test('renders monitoring dashboard title', () => {
    renderWithRouter(<Home />);
    
    expect(screen.getByText('Monitoring')).toBeInTheDocument();
    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
  });

  test('displays dashboard placeholder message', () => {
    renderWithRouter(<Home />);
    
    expect(screen.getByText('There should be a global dashboard...')).toBeInTheDocument();
  });

  test('renders all navigation links', () => {
    renderWithRouter(<Home />);
    
    // Check for all navigation links
    expect(screen.getByRole('link', { name: /Pipeline Requests/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /Batch Requests/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /Job Requests/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /Execution System View/i })).toBeInTheDocument();
  });

  test('navigation links have correct href attributes', () => {
    renderWithRouter(<Home />);
    
    expect(screen.getByRole('link', { name: /Pipeline Requests/i })).toHaveAttribute('href', '/pipeline-request/list');
    expect(screen.getByRole('link', { name: /Batch Requests/i })).toHaveAttribute('href', '/batch-request/list');
    expect(screen.getByRole('link', { name: /Job Requests/i })).toHaveAttribute('href', '/job-request/list');
    expect(screen.getByRole('link', { name: /Execution System View/i })).toHaveAttribute('href', '/execution-system-view');
  });

  test('renders with correct CSS classes for layout', () => {
    renderWithRouter(<Home />);
    
    // Check for responsive grid classes
    const rows = screen.getAllByRole('generic').filter(el => el.className.includes('row'));
    expect(rows).toHaveLength(2);
    
    // Check for column classes
    const cols = screen.getAllByRole('generic').filter(el => el.className.includes('col-'));
    expect(cols.length).toBeGreaterThan(0);
  });

  test('displays Font Awesome icons', () => {
    renderWithRouter(<Home />);
    
    // Check for icon classes
    const icons = document.querySelectorAll('i.fa');
    expect(icons).toHaveLength(4); // 4 navigation items with icons
    
    // Check specific icon classes
    expect(document.querySelector('i.fa-plug')).toBeInTheDocument();
    expect(document.querySelector('i.fa-tasks')).toBeInTheDocument(); 
    expect(document.querySelector('i.fa-rocket')).toBeInTheDocument();
    expect(document.querySelector('i.fa-server')).toBeInTheDocument();
  });

  test('has proper accessibility structure', () => {
    renderWithRouter(<Home />);
    
    // Check for proper heading structure
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveClass('display-4');
    
    // Check for proper link structure
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(4);
    
    links.forEach(link => {
      expect(link).toHaveClass('app-link');
    });
  });

  test('matches snapshot', () => {
    const { container } = renderWithRouter(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});