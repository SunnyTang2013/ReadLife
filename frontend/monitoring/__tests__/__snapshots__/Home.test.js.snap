// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<div>
  <h2
    class="display-4"
  >
    Monitoring
  </h2>
  <section>
    <div
      class="alert alert-primary"
    >
      There should be a global dashboard...
    </div>
  </section>
  <div
    class="row my-4"
  >
    <div
      class="col-12 col-sm-4 col-md-3 col-xl-2 text-center"
    >
      <a
        class="app-link"
        data-discover="true"
        href="/pipeline-request/list"
      >
        <i
          class="app-link-icon teal fa fa-fw fa-plug"
        />
        <span
          class="app-link-title"
        >
          Pipeline Requests
        </span>
      </a>
    </div>
    <div
      class="col-12 col-sm-4 col-md-3 col-xl-2 text-center"
    >
      <a
        class="app-link"
        data-discover="true"
        href="/batch-request/list"
      >
        <i
          class="app-link-icon green fa fa-fw fa-tasks"
        />
        <span
          class="app-link-title"
        >
          Batch Requests
        </span>
      </a>
    </div>
    <div
      class="col-12 col-sm-4 col-md-3 col-xl-2 text-center"
    >
      <a
        class="app-link"
        data-discover="true"
        href="/job-request/list"
      >
        <i
          class="app-link-icon blue fa fa-fw fa-rocket"
        />
        <span
          class="app-link-title"
        >
          Job Requests
        </span>
      </a>
    </div>
  </div>
  <div
    class="row my-4"
  >
    <div
      class="col-12 col-sm-4 col-md-3 col-xl-2 text-center"
    >
      <a
        class="app-link"
        data-discover="true"
        href="/execution-system-view"
      >
        <i
          class="app-link-icon blue fa fa-fw fa-server"
        />
        <span
          class="app-link-title"
        >
          Execution System View
        </span>
      </a>
    </div>
  </div>
</div>
`;
